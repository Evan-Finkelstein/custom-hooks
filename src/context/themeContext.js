import React, { createContext, useState, useContext } from 'react';


export const ThemeContext = createContext(null);


export const ThemeProvider = ({ children }) => {
    const [theme, setTheme] = useState('light');
    return (
        <ThemeContext.Provider value={{ theme, setTheme }}>
            {children}
        </ThemeContext.Provider>
    );
};
export const useTheme = () => {
    const { theme } = useContext(ThemeContext);
    const { setTheme } = useContext(ThemeContext);
    return { theme, setTheme };

};

