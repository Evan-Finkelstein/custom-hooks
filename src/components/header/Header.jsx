import React from 'react';
import { Link } from 'react-router-dom'
import { useTheme } from '../../context/themeContext';
import styles from './Header.css'



const Header = () => {
    const { theme, setTheme } = useTheme()

    const toggleTheme = ({ target }) => {
        if (target.checked) setTheme('dark');
        else setTheme('light')
    }


    return (
        <div className={`${styles[theme]}`}>
            Dark Mode  <input type='checkbox' onChange={toggleTheme} />
            <Link to={'/'}>Home</Link>
        </div>)
}
    ;


export default Header;

