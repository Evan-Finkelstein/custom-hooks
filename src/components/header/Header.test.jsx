import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '../../context/themeContext'


describe('Header component', () => {
  afterEach(() => cleanup());
  it('renders Header', () => {
    const { asFragment } = render(<ThemeProvider><Router><Header /></Router></ThemeProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
