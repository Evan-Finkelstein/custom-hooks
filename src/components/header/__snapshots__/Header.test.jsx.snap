// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <div
    class="light"
  >
    Dark Mode  
    <input
      type="checkbox"
    />
    <a
      href="/"
    >
      Home
    </a>
  </div>
</DocumentFragment>
`;
